src/orient/rational-calc.orient

```diff
@@ -0,0 +1,354 @@
+
+    describe(labeling_proof, "Constraints to check labeling of a single node")
+    describe(labeling_proofs, "Constraints to check labeling of a window")
+    describe(window_labeling_proof, "Constraints to check correct labeling for all windows")
+    describe(data_inclusion, "Constraints to check a single inclusion in CommD")
+    describe(data_inclusions, "Constraints to check all the inclusions required in a window in CommD")
+    describe(window_data_inclusions,  "Constraints to check all the inclusions required for all windows in CommD")
+    describe(replica_inclusion, "Constraints to check a single inclusion in CommR")
+    describe(replica_inclusions, "Constraints to check all the inclusions required in a window in CommR")
+    describe(window_replica_inclusions,  "Constraints to check all the inclusions required for all windows in CommR")
+    describe(window_inclusion, "Constraints to check a single inclusion proof in a window")
+    describe(window_inclusions, "Constraints to check all the required inclusion proofs in a window")
+    describe(column_leaf, "Constraints to check a single leaf generation from a column in a window")
+    describe(column_leaves, "Constraints to check all the leaves generation required in a window")
+    describe(window_column_leaves, "Constraints to check all the leaves generation required for all windows")
+
+    // D
+    // Data layer committed with merkle tree
+    data_inclusion = merkle_tree_datahash_constraints * log2(nodes)
+    data_inclusions = data_inclusion * porep_challenges
+    window_data_inclusions = data_inclusion * porep_challenges * windows
+
+    // C
+    window_inclusion = merkle_tree_hash_constraints * log2(window_nodes)
+    window_inclusions =  window_inclusion * porep_challenges * (graph_parents + 1)
+    column_leaf = column_leaf_hash_constraints * drg_layers
+    column_leaves = column_leaf * porep_challenges * (graph_parents + 1)
+    window_column_leaves = column_leaves * windows
+    labeling_proof = kdf_constraints * graph_parents
+    labeling_proofs = labeling_proof * drg_layers * porep_challenges
+    window_labeling_proof = labeling_proofs * windows
+
+    // Q and R
+    replica_inclusion = merkle_tree_hash_constraints * log2(nodes)
+    replica_inclusions = replica_inclusion * porep_challenges
+    window_replica_inclusions = replica_inclusions * windows
+
+    porep_snark_partitions = porep_snark_constraints / porep_snark_partition_constraints
+    porep_snark_size = porep_snark_partitions * snark_size
+    porep_snark_size_kb = porep_snark_size / 1024
+
+    Constraints (WindowEncoding):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + window_data_inclusions + window_replica_inclusions*2
+
+    Constraints (SectorEncoding):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + data_inclusions + replica_inclusions*2
```

Comment:
Should there be a factor of `windows` in here somewhere?

Reply:
each of those should account for it.